rest/src/main/java/fi/otavanopisto/pyramus/rest/controller/StudentController.java

```diff
@@ -537,6 +542,33 @@ public void deleteLodgingPeriod(StudentLodgingPeriod period) {
 
     return result;
   }
+  
+  /* Checks if logged user is guidance counselor of specific student */
+  public boolean amIGuidanceCouncelor(Long studentId, StaffMember staffMember) {
+    
+    Student student = findStudentById(studentId);
+    // List of student's user groups
+    List<StudentGroup> studentGroups = studentGroupDAO.listByStudent(student, 0, 50, false);
+    
+    for (StudentGroup studentGroup : studentGroups) {
+      Set<StudentGroupUser> studentGroupUsers = studentGroup.getUsers();
+      
+      if (!studentGroupUsers.isEmpty()) {
+        if (staffMember == null) {
```

Comment:
This kind of check should really be done at the very beginning of the method. Why even bother listing groups first if `staffMember` is `null`?